src/basic/string-util.c

```diff
@@ -1133,3 +1133,66 @@ int string_contains_word_strv(const char *string, const char *separators, char *
                 *ret_word = found;
         return !!found;
 }
+
+static bool is_digit(int c) {
+        return c >= '0' && c <= '9';
+}
+
+static int c_order(int c) {
+        if (c == 0 || is_digit(c))
+                return 0;
+
+        if ((c >= 'a') && (c <= 'z'))
+                return c;
+
+        return c + 0x10000;
+}
+
+int str_verscmp(const char *s1, const char *s2) {
```

Comment:
we should probably find a different name for the function, it's just confusing that there's strverscmp() and str_verscmp() and they do the same thing but also quite differently.

maybe rename this strverscmp_improved() or strverscmp_extra() or strverscmp_with_a_twist() or so.. ;-)